datafusion/expr/src/expr_schema.rs

```diff
@@ -762,29 +806,25 @@ mod tests {
 
     #[derive(Debug)]
     struct MockExprSchema {
-        nullable: bool,
-        data_type: DataType,
+        field: Field,
```

Comment:
the fact this is simpler but now also more full featured suggests to me we are on the right track